types/next-auth/index.d.ts

```diff
@@ -0,0 +1,174 @@
+// Type definitions for next-auth 3.1
+// Project: https://github.com/iaincollins/next-auth#readme
+// Definitions by: Lluis <https://github.com/lluia>
+//                 Iain <https://github.com/iaincollins>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.1
+
+/// <reference types="node" />
+
+import { ConnectionOptions } from 'typeorm';
+import { PossibleProviders } from './providers';
+import { Adapter } from './adapters';
+
+interface InitOptions {
```

Comment:
Would it be possible to export this interface? I'm trying these types out locally on a new project using NextAuth, and for reasons that I don't totally grok, the TS compiler seems to struggle to infer the correct `ConnectionOptions` type if I don't define my `options` object as being of type `InitOptions` up front. That is, this doesn't work:

```typescript
const options = {
  database: {
    type: 'postgres',
    host: process.env.DB_HOST,
    port: parseInt(process.env.DB_PORT, 10),
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_DATABASE,
  },
}

export default (req, res) => NextAuth(req, res, options)
```

Because the compiler complains that:

> Type '{ type: string; host: string; port: number; username: string; password: string; database: string; }' is missing the following properties from type 'AuroraDataApiConnectionOptions': region, secretArn, resourceArn

But this is fine: 

```typescript
const options: InitOptions = {
  database: {
    type: 'postgres',
    host: process.env.DB_HOST,
    port: parseInt(process.env.DB_PORT, 10),
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_DATABASE,
  },
}

export default (req, res) => NextAuth(req, res, options)
```

Reply:
@SMores yup definitively fine having those ones exported 👍🏽 , let me amend this in the afternoon 🗓

Reply:
@SMores addressed, if you can  👀 again plz